Neos.ContentRepository.Core/Classes/Infrastructure/DbalCheckpointStorage.php

```diff
@@ -115,6 +115,9 @@ public function updateAndReleaseLock(SequenceNumber $sequenceNumber): void
         if (!$this->connection->isTransactionActive()) {
             throw new \RuntimeException(sprintf('Failed to update and commit checkpoint for subscriber "%s" because no transaction is active', $this->subscriberId), 1652279314);
         }
+        if ($this->connection->isRollbackOnly()) {
+            throw new \RuntimeException(sprintf('Failed to update and commit checkpoint for subscriber "%s" because the transaction has been marked for rollback only', $this->subscriberId), 1711964313);
+        }
```

Comment:
```suggestion
        if ($this->connection->isRollbackOnly()) {
            // TODO as described in https://github.com/neos/neos-development-collection/issues/4970 we are in a bad state and cannot commit after a nested transaction was rolled back.
            throw new \RuntimeException(sprintf('Failed to update and commit checkpoint for subscriber "%s" because the transaction has been marked for rollback only. See https://github.com/neos/neos-development-collection/issues/4970', $this->subscriberId), 1711964313);
        }
```